modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+
+      if (initialize_me)
+      {
+        params.set<VariableName>("variable") = _passive_scalar_names[name_i];
+        if (isParamValid("initial_scalar_variables"))
+          params.set<FunctionName>("function") =
+              getParam<std::vector<FunctionName>>("initial_scalar_variables")[ic_counter];
+        else
+          params.set<FunctionName>("function") = "0.0";
+
+        _problem->addInitialCondition("FunctionIC", _passive_scalar_names[name_i] + "_ic", params);
+        ic_counter += 1;
+      }
+    }
+  }
+}
+
+void
+NSFVAction::addINSMomentumTimeKernels()
+{
+  std::string kernel_type = "INSFVMomentumTimeDerivative";
+  std::string kernel_name = "ins_momentum_time_";
+  std::string rhie_chow_name = "ins_rhie_chow_interpolator";
+  const std::string * velocity_vector_name = NS::velocity_vector;
+
+  if (_porous_medium_treatment)
+  {
+    kernel_type = "PINSFVMomentumTimeDerivative";
+    kernel_name = "pins_momentum_time_";
+    rhie_chow_name = "pins_rhie_chow_interpolator";
+    velocity_vector_name = NS::superficial_velocity_vector;
+  }
+
+  InputParameters params = _factory.getValidParams(kernel_type);
+  params.set<std::vector<SubdomainName>>("block") = _blocks;
+  params.set<MooseFunctorName>(NS::density) = _density_name;
+  params.set<UserObjectName>("rhie_chow_user_object") = rhie_chow_name;
+
+  for (unsigned int d = 0; d < _dim; ++d)
+  {
+    params.set<NonlinearVariableName>("variable") = velocity_vector_name[d];
+    params.set<MooseEnum>("momentum_component") = NS::directions[d];
+
+    _problem->addFVKernel(kernel_type, kernel_name + NS::directions[d], params);
+  }
+}
+
+void
+NSFVAction::addINSEnergyTimeKernels()
+{
+  std::string kernel_type = "INSFVEnergyTimeDerivative";
+  std::string kernel_name = "ins_energy_time";
+
+  if (_porous_medium_treatment)
+  {
+    kernel_type = "PINSFVEnergyTimeDerivative";
+    kernel_name = "pins_energy_time";
+  }
+
+  InputParameters params = _factory.getValidParams(kernel_type);
+  params.set<std::vector<SubdomainName>>("block") = _blocks;
+  params.set<NonlinearVariableName>("variable") = NS::T_fluid;
+  params.set<MooseFunctorName>(NS::density) = _density_name;
+  params.set<MooseFunctorName>(NS::cp) = _specific_heat_name;
+
+  if (_porous_medium_treatment)
+  {
+    params.set<MooseFunctorName>(NS::porosity) = _porosity_name;
+    if (_problem->hasFunctor(NS::time_deriv(NS::density),
+                             _problem->parameters().get<THREAD_ID>("_tid")))
+      params.set<MooseFunctorName>(NS::time_deriv(NS::density)) = NS::time_deriv(_density_name);
+    if (_problem->hasFunctor(NS::time_deriv(_specific_heat_name),
+                             _problem->parameters().get<THREAD_ID>("_tid")))
+      params.set<MooseFunctorName>(NS::time_deriv(NS::cp)) = NS::time_deriv(_specific_heat_name);
+    params.set<bool>("is_solid") = false;
+  }
+  else
+    params.set<MooseFunctorName>(NS::time_deriv(NS::cp)) = NS::time_deriv(_specific_heat_name);
```

Comment:
add the functor being constant
this dies with
```
No functor ever provided with name 'd1594_dt', which was requested by 'ins_energy_time'.
```
for constant functors. let s pass 0 for constants ?